pkg/storage/chunk/client/object_client.go

```diff
@@ -105,6 +106,15 @@ func (o *client) Stop() {
 	o.store.Stop()
 }
 
+func (o *client) PutWal(ctx context.Context, segment *wal.SegmentWriter) error {
+	buffer := bytes.NewBuffer(nil)
+	_, err := segment.WriteTo(buffer)
+	if err != nil {
+		return err
+	}
+	return o.store.PutObject(ctx, "wal-segment-"+time.Now().UTC().Format(time.RFC3339Nano), bytes.NewReader(buffer.Bytes()))
```

Comment:
Can the new ingester deal with this interface instead ?  `PutObject`

This way we don't leak our storage into object storage clients.

Reply:
I think thats a good idea but for now I was just copying the existing interfaces which use "PutChunks" everywhere. There is a slightly lower level interface which only has PutObject, but our storage layer seems to switch between them frequently.

Reply:
I addressed it in https://github.com/grafana/loki/pull/13342/commits/b4ca98be8441f09f095f6edc44648ffcc97e54b3